doc/config.rst

```diff
@@ -71,3 +71,20 @@ configure them in your config file.
         ->setIndent("\t")
         ->setLineEnding("\r\n")
     ;
+
+It's possible to register custom rule sets, which makes it easier to reuse custom configuration between multiple projects. If you have prepared rule set, you can register it and then enable it in the rules. Custom rule sets (in this example `\MyNameSpace\MyRuleSetClass`) must implement `\PhpCsFixer\RuleSet\RuleSetDescriptionInterface`.
```

Comment:
First of all, big thanks @niklam to take effort on this PR, and thanks @Wirone to take it back from dead and collaborating with @niklam to move this PR further 🍻 !

Yet here, before next code changes, I would love to align on the approach to how we would like to achieve the custom rulesets first.